services/agents/message_dispatcher.go

```diff
@@ -0,0 +1,38 @@
+package agents
+
+import (
+	"sync"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/percona/pmm/api/agent"
+)
+
+type messageDispatcher struct {
+	sync.RWMutex
```

Comment:
Never embed `sync.Mutex` or `sync.RWMutex` – that makes `Lock()` / `Unlock()` methods a part of exported API.
```go
rw sync.RWMutex
```